extension/src/api.rs

```diff
@@ -285,3 +289,177 @@ fn chunk_text(document: &str, max_characters: i64) -> Vec<String> {
+                return Err(anyhow::anyhow!(
+                    "Row {} has embedding length {} but expected {}",
+                    original_id,
+                    embeddings.len(),
+                    expected_dim
+                ));
+            }
+
+            let insert_q = format!(
+                "INSERT INTO vectorize._embeddings_{} ({}, embeddings, updated_at)
+                 VALUES ($1, $2, NOW())
+                 ON CONFLICT ({}) DO UPDATE SET embeddings = $2, updated_at = NOW()",
+                job_name, job_params.primary_key, job_params.primary_key
+            );
+
+            Spi::run_with_args(
+                &insert_q,
+                Some(vec![
+                    (PgBuiltInOids::TEXTOID.oid(), original_id.into_datum()),
+                    (PgBuiltInOids::FLOAT8ARRAYOID.oid(), embeddings.into_datum()),
+                ]),
+            )?;
+            count += 1;
+        }
+    } else {
+        // For append method, update the source table's embeddings column
+        let update_q = format!(
+            "UPDATE {}.{} t0 
+             SET {}_embeddings = src.{},
+                 {}_updated_at = NOW()
+             FROM {} src
+             WHERE t0.{} = src.{}",
+            job_params.schema,
+            job_params.table,
+            job_name,
+            src_embeddings_col,
+            job_name,
+            src_table,
+            job_params.primary_key,
+            src_primary_key
+        );
+
+        Spi::run(&update_q)?;
+        count = Spi::get_one::<i64>("SELECT count(*) FROM pg_temp.pg_stat_statements_info")?
+            .unwrap_or(0) as i32;
+    }
+
+    // Clean up realtime jobs if necessary
+    if job_params.schedule == "realtime" {
+        let delete_q = format!(
+            "DELETE FROM pgmq.q_{} WHERE message->>'job_name' = $1",
+            VECTORIZE_QUEUE
+        );
+        Spi::run_with_args(
+            &delete_q,
+            Some(vec![(PgBuiltInOids::TEXTOID.oid(), job_name.into_datum())]),
+        )?;
+    }
+
+    Ok(format!(
+        "Successfully imported embeddings for {} row(s)",
+        count
+    ))
+}
+
+#[allow(clippy::too_many_arguments)]
+#[pg_extern]
+fn table_from(
+    table: &str,
+    columns: Vec<String>,
+    job_name: &str,
+    primary_key: &str,
+    src_table: &str,
+    src_primary_key: &str,
+    src_embeddings_col: &str,
+    schema: default!(&str, "'public'"),
+    update_col: default!(String, "'last_updated_at'"),
+    index_dist_type: default!(types::IndexDist, "'pgv_hnsw_cosine'"),
+    transformer: default!(&str, "'sentence-transformers/all-MiniLM-L6-v2'"),
+    table_method: default!(types::TableMethod, "'join'"),
+    schedule: default!(&str, "'* * * * *'"),
+) -> Result<String> {
+    let model = Model::new(transformer)?;
+
+    // First initialize the table structure without triggers/cron
+    init_table(
+        job_name,
+        schema,
+        table,
+        columns,
```

Comment:
For now I think it will be fine to just clone here. That will help get your tests to pass :) 

```suggestion
        columns.clone(),
```